lib/pure/collections/lists.nim

```diff
@@ -669,13 +712,36 @@ proc add*[T: SomeLinkedList](a: var T, b: T) {.since: (1, 5, 1).} =
   var tmp = b.copy
   a.addMoved tmp
 
+proc remove*[T](L: var SinglyLinkedList[T], n: SinglyLinkedNode[T]) =
+  ## Removes a node `n` from `L`.
+  ## Efficiency: O(n), unless `n` is the head element.
```

Comment:
` unless `n` is the head element.` doesn't really make sense in the context of discussing complexity, and also is not very useful because runtime is also low if element appears early in list; instead do this:
```suggestion
  ## Efficiency: O(n); the list is traversed until `n` is found.
```